odkx-src/injected-interfaces.rst

```diff
@@ -733,3 +786,520 @@ removeFirstQueuedAction
+The function uses the ``getOdkDataIf().changeAccessFilterOfRow`` method to update the access control settings for the row.
+Callback functions are registered to handle the success or failure of the access control change operation.
+
+updateRow
+~~~~~~~~~~~
+
+  **Parameters**:
+
+  - tableId: The ID of the table to which the row belongs.
+  - columnNameValueMap: A JSON object representing the column name to new value mapping for the row.
+  - rowId: The ID of the specific row to be updated.
+  - successCallbackFn: A success callback function that is called when the row update operation is successful.
+  - failureCallbackFn: A failure callback function that is called when there is an error in updating the row.
+
+This function is used to update a specific row in a table with new values for the specified columns.
+It takes a JSON object ``columnNameValueMap`` that maps column names to their new values.
+The function uses the ``getOdkDataIf().updateRow`` method to update the row with the provided values.
+Callback functions are registered to handle the success or failure of the update operation.
+
+deleteRow
+~~~~~~~~~~
+
+  **Parameters**:
+
+  - tableId: The ID of the table from which the row should be deleted.
+  - columnNameValueMap: A JSON object representing the column name to value mapping for identifying the row to be deleted.
+  - rowId: The ID of the specific row to be deleted.
+  - successCallbackFn: A success callback function that is called when the row deletion operation is successful.
+  - failureCallbackFn: A failure callback function that is called when there is an error in deleting the row.
+
+This function is used to delete a specific row from a table.
+It takes a JSON object ``columnNameValueMap`` that maps column names to their values, and a ``rowId`` to identify the row to be deleted.
+The function uses the ``getOdkDataIf().deleteRow`` method to delete the row based on the provided criteria.
+Callback functions are registered to handle the success or failure of the delete operation.
+
+addRow
+~~~~~~~
+
+  Parameters:
+
+  - tableId: The ID of the table to which the new row should be added.
+  - columnNameValueMap: A JSON object representing the column name to value mapping for the new row.
+  - rowId: The ID for the new row.
+  - successCallbackFn: A success callback function that is called when the row addition operation is successful.
+  - failureCallbackFn: A failure callback function that is called when there is an error in adding the row.
+
+This function is used to add a new row to a table.
+It takes a JSON object ``columnNameValueMap`` that maps column names to their values for the new row.
+The ``rowId`` is used to specify the ID of the new row.
+The function uses the ``getOdkDataIf().addRow`` method to add the new row to the specified table.
+Callback functions are registered to handle the success or failure of the addition operation.
+
+addCheckpoint
+~~~~~~~~~~~~~~
+
+  **Parameters**:
```

Comment:
May want to also add somewhere in here a cross-link to explain checkpoint saves.